robot-core/src/main/java/org/obolibrary/robot/metrics/MetricsUtils.java

```diff
@@ -0,0 +1,239 @@
+public class MetricsUtils {
+
+  /**
+   * @param axioms set of axioms
+   * @return filtered set of axioms (only ABox axioms)
+   */
+  public static Set<OWLAxiom> getABoxAxioms(Set<OWLAxiom> axioms) {
+    Set<OWLAxiom> abox = new HashSet<>();
+    Set<AxiomType<?>> types = getABoxAxiomTypes();
+    for (OWLAxiom ax : axioms) {
+      if (types.contains(ax.getAxiomType())) {
+        abox.add(ax);
+      }
+    }
+    return abox;
+  }
+
+  /**
+   * @param ontology ontology to be checked
+   * @param includeImportsClosure take into account imports
+   * @param skiprules ignore rules (SWRL)
+   * @param types AxiomTypes to consider
+   * @return all logical axioms in the ontology
+   */
+  public static Set<OWLAxiom> getLogicalAxioms(
+      OWLOntology ontology,
+      Imports includeImportsClosure,
+      boolean skiprules,
+      Set<AxiomType<?>> types) {
+    return getLogicalAxioms(ontology, includeImportsClosure, skiprules, false, types);
+  }
+
+  /**
+   * @param including_rbox whether or not RBox
+   * @return all TBox axiom types
+   */
+  public static Set<AxiomType<?>> getTBoxAxiomTypes(boolean including_rbox) {
+    Set<AxiomType<?>> axty = new HashSet<>();
+    if (including_rbox) {
+      axty.addAll(AxiomType.RBoxAxiomTypes);
+    }
+    axty.addAll(AxiomType.TBoxAxiomTypes);
+    return axty;
+  }
+
+  /**
+   * @param axioms set of axioms
+   * @return filtered set of axioms (only tbox axioms)
+   */
+  public static Set<OWLAxiom> getTBoxAxioms(Set<OWLAxiom> axioms) {
+    Set<OWLAxiom> tbox = new HashSet<>();
+    Set<AxiomType<?>> types = getTBoxAxiomTypes(true);
+    for (OWLAxiom ax : axioms) {
+      if (types.contains(ax.getAxiomType())) {
+        tbox.add(ax);
+      }
+    }
+    return tbox;
+  }
+
+  private static Set<OWLAxiom> stripAnnotations(Set<OWLAxiom> axioms) {
+    Set<OWLAxiom> newAxioms = new HashSet<>();
+    for (OWLAxiom ax : axioms) {
+      if (ax instanceof OWLAnnotationAssertionAxiom) {
+        continue;
+      }
+      newAxioms.add(ax.getAxiomWithoutAnnotations());
+    }
+    return newAxioms;
+  }
+
+  private static boolean isIntegerParsable(String id) {
+    try {
+      Integer.parseInt(id);
+      return true;
+    } catch (NumberFormatException e) {
+      return false;
+    }
+  }
+
+  private static boolean isLongParsable(String id) {
+    try {
+      Long.parseLong(id);
+      return true;
+    } catch (NumberFormatException e) {
+      return false;
+    }
+  }
+
+  private static boolean isDoubleParsable(String id) {
+    try {
+      Double d = Double.parseDouble(id);
+      return !d.isNaN();
+    } catch (NumberFormatException e) {
+      return false;
+    }
+  }
+
+  private static boolean isNumeric(String id) {
+    return (isIntegerParsable(id) || isLongParsable(id) || isDoubleParsable(id));
+  }
+
+  private static Set<OWLAxiom> getLogicalAxioms(
+      OWLOntology ontology,
+      Imports includeImportsClosure,
+      boolean skiprules,
+      boolean leavedeclarations) {
```

Comment:
> Actual value of parameter 'leavedeclarations' is always 'false'

Is this something that we need to add an option to the command for? Since this is private, it's never accessed anywhere else.

Reply:
I think in this case, I would just like to keep the parameter, as I may want to change this behaviour in the future, and this reminds me that this actually was a _thing_. So each time I write code and use this method, I will be reminded to consider this stupid problem: 

Declaration(A) -> kinda like A subclass of owl:Thing, but its not strictly speaking logical - so you may lose signature if you look at the logical axioms without considering the Declarations.